src/Numerics/Mesh/FTP.jl

```diff
@@ -0,0 +1,201 @@
+using DocStringExtensions
+using KernelAbstractions
+using CUDA
+
+include("FTP_kernels.jl")
+
+"""
+    ftpxv!(grd::DiscontinuousSpectralElementGrid,
+        mesh::Symbol,
+        opts::Symbol,
+        tropts::Bool,
+        vin::AbstractArray{FT},
+        vout::AbstractArray{FT},
+        vin_den::Union{AbstractArray{FT}, Nothing} = nothing,
+    ) where {FT<:AbstractFloat}
+
+This function computes the inner derivatives of a field using the fast
+tensor-product algorithm on a given mesh. Currently, DG mesh (`:m1`) and
+over-integration mesh (`:m2`) are supported. The function provides the 
+building blocks for constructing various terms in the DG formulation and 
+for constructing some diagnostic fields.
+
+# Arguments
+ - `grd`: DiscontinuousSpectralElementGrid
+ - `mesh`: quadrature mesh (currently, `:m1` and `:m2` supported)
+ - `opts`: inner derivative to be computed (currently, `:ξ1`, `:ξ2`, `:ξ3` supported)
+ - `tropts`: transpose (true/false)
```

Comment:
what does this do in practice?

Reply:
The fuction contains all combinations of fast tensor-product needed for contructing all the volume terms, either with CG or DG.